### PR TITLE
Updated FaF framework dependencies and made a few performance tweaks

### PR DESCRIPTION
**Pull request by errantmind**

### CODE REVIEW

**Comment by errantmind** [comment]
Hello @nbrady-techempower, any chance this can make it in before the next test cycle (~24h from now)? I'm curious about some of my performance improvements.

File: frameworks/Rust/faf/src/main.rs [hunk 1]
```diff
@@ -35,55 +30,43 @@ const PLAINTEXT_BASE: &[u8] = const_concat_bytes!(
...
-   _headers: &[faf::phr_header; MAX_HEADERS_TO_PARSE],
-   _num_headers: usize,
    response_buffer: *mut u8,
+   date_buff: *const u8,
 ) -> usize {
    unsafe {
       if likely(method_len >= GET_LEN && path_len >= ROUTE_PLAINTEXT_LEN) {
-         if likely(memcmp(GET.as_ptr() as *const i8, method, GET_LEN) == 0) {
+         if likely(memcmp(GET.as_ptr(), method, GET_LEN) == 0) {
             // For performance purposes, this will successfully match '/p' to '/plaintext' and '/pickle'. Use with caution
```
Comment (nbrady-techempower) [inline frameworks/Rust/faf/src/main.rs:48]: I've missed this before, and I'm not sure how many other people are doing it, but I don't like this. I want it to only match `/plaintext` but we're not enforcing that.

Comment (errantmind) [reply frameworks/Rust/faf/src/main.rs]: I updated the code to perform a 'full' comparison. It should be good to go now

**Comment by errantmind** [comment]
Thanks for getting this merged before the next test cycle :+1:

**Pull request merged**